### PR TITLE
Fixing merge conflict issues in release-4.1.0

### PR DESCRIPTION
**Pull request by paul-graham**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/projects/EUI/versions/39208
https://tools.hmcts.net/jira/projects/EUI/versions/38816

### Change description ###
This PR is specifically to address the issues that were identified in https://github.com/hmcts/ccd-case-ui-toolkit/pull/819, which must have come about when rebasing `release-4.1.0` against `master`. It appears that local changes were kept in favour of incoming changes from `master` whenever there was a conflict, which has resulted in a lot of fixes and changes being ditched. It _seems_ like some of the changes were also ditched in files where I can't see how there could have been a conflict as there are no changes going back the other way - just a reversal of what's currently (correctly) on `master`.

Here's a breakdown of what this PR contains:

* All of `Version 4.0.0` was missing from `RELEASE-NOTES.md` so that's been put back in.
* Set the version in `package.json` to be `4.1.0`.
* Reinstated a bunch of tidy-ups in `demo/test` that seemed to have been dismissed.
* Reinstated unit test functionality in the `case-edit-page.component`, which had been incorrectly ditched.
* Re-fixed `getCaseId(): String` to be `public getCaseId(): string` in `case-edit-submit.component`.
* Fixed up a bunch of regressions in the .html files within `shared/components/palette` where `dirty` had been incorrectly replaced with `touched`.
* Re-fixed an issue with labels not showing for collections when there's no `hint_text` and the array is not in error.
* Re-fixed EUI-4227, which had been lost.
* Reinstantiated some changes and fixes in the `FirstErrorPipe`.
* Undid the weird replacement of the `FirstErrorPipe` tests with a copy of the `FirstErrorPipe` class.
* Re-introduced 25 unit tests for `FieldUtils.containsNonEmptyValues(...)`, which had been incorrectly ditched.
* Re-removed pointless whitespace in various places.
* Running "Organise imports" has also done a bit of cleaning up.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by phillip-whitaker — APPROVED** [review]
LGBTQ+

**Review by Daniel-Lam — COMMENTED** [review]
@paul-graham - I've done a thorough look-through, and it seems you've caught all the reversions - looks good. Just a couple of things (which I'll probably look at):

- Removing entries in `RELEASE-NOTES.md` for some of the earlier features with a tag release, which are now condensed under `Version 4.0.0`
- Some of the imports in `case-edit-page.component.spec.ts` can be grouped together
- Reversion of named parameters in `conditional-show-form.directive.ts` (in PR #819) hasn't been undone in this PR

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -1,5 +1,37 @@
 ## RELEASE NOTES
 
+### Version 4.0.0
```
[the remark this answers is not in the document]
Comment (paul-graham) [reply RELEASE-NOTES.md:17]: I've since added an entry for `Version 4.1.0` too, @Daniel-Lam.

**Review by Daniel-Lam — APPROVED** [review]
@paul-graham - all looks good; approved.

**Pull request merged**